app/controllers/mixins/ems_common_angular.rb

```diff
@@ -69,7 +73,7 @@ def update_ems_button_validate(verify_ems = nil)
         add_flash(_("Credential validation was not successful: %{details}") % {:details => strip_tags(details)}, :error)
       end
 
-      render :json => {:message => @flash_array.last(1)[0][:message], :level => @flash_array.last(1)[0][:level]}
+      render_flash_json
```

Comment:
You are right, we can completely do away with `@flash_array`. It's not needed.
Can be easily done right here in this PR - so no future refactoring :)